nautobot/core/settings_funcs.py

```diff
@@ -82,3 +84,93 @@ def parse_redis_connection(redis_database):
         return f"{redis_scheme}://{redis_creds}{redis_host}?db={redis_database}"
     else:
         return f"{redis_scheme}://{redis_creds}{redis_host}:{redis_port}/{redis_database}"
+
+
+def setup_structlog_logging(
+    django_logging: dict,
+    django_apps: list,
+    django_middleware: list,
+    log_level="INFO",
+    root_level="INFO",
+    debug_db=False,
+    plain_format=False,
+) -> None:
+    """Set up structlog logging for Django."""
+    django_logging["version"] = 1
+    django_logging["disable_existing_loggers"] = True
+    if "test" in sys.argv:
+        django_logging["handlers"] = {
+            "null_handler": {
+                "level": "INFO",
+                "class": "logging.NullHandler",
+            },
+        }
+        for logger in django_logging["loggers"].values():
+            logger["handlers"] = ["null_handler"]
+            logger["level"] = "INFO"
+
+        return
+
+    django_apps.append("django_structlog")
+    django_middleware.append("django_structlog.middlewares.RequestMiddleware")
+
+    processors = (
+        # Add the log level to the event dict under the level key.
+        structlog.stdlib.add_log_level,
+        # Add the logger name to the event dict.
+        structlog.stdlib.add_logger_name,
+        # Apply stdlib-like string formatting to the event key.
+        structlog.stdlib.PositionalArgumentsFormatter(),
+        # A processor that merges in a global (context-local) context.
+        structlog.contextvars.merge_contextvars,
+        # Add a timestamp to event_dict in ISO 8601 format.
+        structlog.processors.TimeStamper(fmt="iso"),
+        # Replace an exc_info field with an exception string field using Python's built-in traceback formatting.
+        structlog.processors.format_exc_info,
+        # Add stack information with key stack if stack_info is True.
+        structlog.processors.StackInfoRenderer(),
+        # Decode byte string values in event_dict.
+        structlog.processors.UnicodeDecoder(),
+    )
+
+    django_logging["formatters"] = {
+        "default_formatter": {
+            "()": structlog.stdlib.ProcessorFormatter,
+            "foreign_pre_chain": processors,
+            "processor": structlog.dev.ConsoleRenderer() if plain_format else structlog.processors.JSONRenderer(),
+        },
+    }
```

Comment:
Should this add to the existing `formatters` dict instead of overwriting it entirely?

```suggestion
    django_logging["formatters"]["default_formatter"] = {
        "()": structlog.stdlib.ProcessorFormatter,
        "foreign_pre_chain": processors,
        "processor": structlog.dev.ConsoleRenderer() if plain_format else structlog.processors.JSONRenderer(),
    }
```

Reply:
ad Overwrite vs add formatters and handlers:

If we change this to add strategy, then it leads to duplicit logs with the default config. IMHO it's better to overwrite all handlers and filters if exists. User then can selectively add handlers, formatters or loggers  after calling this function. Leads to a cleaner usage pattern.

Reply:
Added comments to the settings, and function docstring.